enhancements/builds/remove-jenkins-playload.md

```diff
@@ -0,0 +1,195 @@
+---
+title: remove-jenkins-payload
+authors:
+  - "@adambkaplan"
+reviewers:
+  - "@gabemontero"
+  - "@akram"
+approvers:
+  - "@bparees"
+  - "@sbose78"
+  - "@derekwaynecarr"
+creation-date: 2021-07-20
+last-updated: 2021-07-20
+status: provisional
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Remove Jenkins from the OCP Payload
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This proposal aims to remove Jenkins images from the OCP payload and replace them with images delivered via CPaaS.
+
+## Motivation
+
+Distrbuting and supporting Jenkins directly has always been challenging for OpenShift.
+Due to OpenShift's tight coupling to Jenkins via the Jenkins Pipeline build strategy,
+OpenShift 4 has included a Jenkins distribution as well as a default set of agent images used to run Jenkins pipelines.
+This meant that any bug fixes or upgrades to Jenkins needed to go through the often arduous backport process to get delivered to customers.
+
+Jenkins itself is not native to Kubernetes and does not use a Java runtime that has been optimized for cloud-native environments (i.e. Quarkus).
+As a result, it has poor resilience when run on Kubernetes.
+The Jenkins templates do not run Jenkins in a highly available mode, leaving its primary instance vulnerable to downtime from slow restarts.
+The Jenkins container image is also very large - it is in fact the largest image in the OpenShift payload today.
+Removing Jenkins from the OpenShift payload has been a longstanding goal that can reduce the time needed to pull the OCP payload at install/updgrade time.
+
+With the GA release of OpenShift Pipelines, we would like to decouple Jenkins from OpenShift, distributing Jenkins just like any other application.
+This would allow bug fixes in our Jenkins distribution to be released on its own cadence.
+Customers can continue to deploy Jenkins on OpenShift via its ImageStream and associated templates.
+In this fashion, Jenkins becomes just another application on OpenShift and does not retain any preferential status.
+The maintentance of Jenkins is also simplified - only one image version will be released and tested against supported OCP versions.
```

Comment:
```suggestion
The maintenance of Jenkins is also simplified - only one image version will be released and tested against supported OCP versions.
```

Reply:
we tried to ship only a single jenkins image for all ocp versions back in the 3.x days.  It eventually bit us, hard, due to api changes.  I would strongly advise against trying to go down that road again.  I think you're going to need to ship/maintain a jenkins version tag per ocp version.

specifically, when every customer is using the "latest" tag, and you need to make an incompatible change (one that is necessary to support 4.12 for example, but breaks 4.8) you can't do it, unless you can find a way to make the change be ocp version aware and do the right thing in both cases, which may be incredibly difficult particularly if the issue is in some jenkins plugin code or something.

Reply:
At a minimum you should probably publish multiple tags even if they use the same content for now.  And ensure that OCP clusters import the tag that's aligned to their OCP version.  That way if you *do* need to break compatibility in the future, you can just branch and start publishing 2 different images at that point.